ros/src/computing/planning/motion/packages/op_local_planner/nodes/op_motion_predictor/op_motion_predictor_core.cpp

```diff
@@ -0,0 +1,310 @@
+		_nh.getParam("/op_common_params/rollOutsNumber", m_PlanningParams.rollOutNumber);
+	else
+		m_PlanningParams.rollOutNumber = 0;
+
+	std::cout << "Rolls Number: " << m_PlanningParams.rollOutNumber << std::endl;
+
+	_nh.getParam("/op_common_params/horizonDistance", m_PlanningParams.horizonDistance);
+	_nh.getParam("/op_common_params/minFollowingDistance", m_PlanningParams.minFollowingDistance);
+	_nh.getParam("/op_common_params/minDistanceToAvoid", m_PlanningParams.minDistanceToAvoid);
+	_nh.getParam("/op_common_params/maxDistanceToAvoid", m_PlanningParams.maxDistanceToAvoid);
+	_nh.getParam("/op_common_params/speedProfileFactor", m_PlanningParams.speedProfileFactor);
+
+	_nh.getParam("/op_common_params/horizontalSafetyDistance", m_PlanningParams.horizontalSafetyDistancel);
+	_nh.getParam("/op_common_params/verticalSafetyDistance", m_PlanningParams.verticalSafetyDistance);
+
+	_nh.getParam("/op_common_params/enableLaneChange", m_PlanningParams.enableLaneChange);
+
+	_nh.getParam("/op_common_params/width", m_CarInfo.width);
+	_nh.getParam("/op_common_params/length", m_CarInfo.length);
+	_nh.getParam("/op_common_params/wheelBaseLength", m_CarInfo.wheel_base);
+	_nh.getParam("/op_common_params/turningRadius", m_CarInfo.turning_radius);
+	_nh.getParam("/op_common_params/maxSteerAngle", m_CarInfo.max_steer_angle);
+	_nh.getParam("/op_common_params/maxAcceleration", m_CarInfo.max_acceleration);
+	_nh.getParam("/op_common_params/maxDeceleration", m_CarInfo.max_deceleration);
+	m_CarInfo.max_speed_forward = m_PlanningParams.maxSpeed;
+	m_CarInfo.min_speed_forward = m_PlanningParams.minSpeed;
+
+	int iSource = 0;
+	_nh.getParam("/op_common_params/mapSource" 			, iSource);
+	if(iSource == 0)
+		m_MapType = PlannerHNS::MAP_AUTOWARE;
+	else if (iSource == 1)
+		m_MapType = PlannerHNS::MAP_FOLDER;
+	else if(iSource == 2)
+		m_MapType = PlannerHNS::MAP_KML_FILE;
+
+	_nh.getParam("/op_common_params/mapFileName" 		, m_MapPath);
+
+	_nh.getParam("/op_motion_predictor/enableGenrateBranches" 		, m_PredictBeh.m_bGenerateBranches);
+	_nh.getParam("/op_motion_predictor/max_distance_to_lane" 		, m_PredictBeh.m_MaxLaneDetectionDistance);
+	_nh.getParam("/op_motion_predictor/prediction_distance" 		, m_PredictBeh.m_PredictionDistance);
+	_nh.getParam("/op_motion_predictor/enableCurbObstacles"			, m_bEnableCurbObstacles);
+
+	UtilityHNS::UtilityH::GetTickCount(m_SensingTimer);
+}
+
+void MotionPrediction::callbackGetCurrentPose(const geometry_msgs::PoseStampedConstPtr& msg)
+{
+	m_CurrentPos = PlannerHNS::WayPoint(msg->pose.position.x, msg->pose.position.y, msg->pose.position.z, tf::getYaw(msg->pose.orientation));
+	bNewCurrentPos = true;
+}
+
+void MotionPrediction::callbackGetVehicleStatus(const geometry_msgs::TwistStampedConstPtr& msg)
+{
+	m_VehicleStatus.speed = msg->twist.linear.x;
+	m_CurrentPos.v = m_VehicleStatus.speed;
+	if(fabs(msg->twist.linear.x) > 0.25)
+		m_VehicleStatus.steer = atan(m_CarInfo.wheel_base * msg->twist.angular.z/msg->twist.linear.x);
```

Comment:
here you guard against division by 0 but what happens then to `m_VehicleStatus.steer`?

Reply:
zero if first call, last value otherwise